packages/driver/src/cypress/cy.ts

```diff
@@ -321,906 +352,868 @@ class $Cy implements ITimeouts, IStability, IAssertions, IRetries, IJQuery, ILoc
-          return Cypress.action('app:window:alert', str)
-        },
-        onConfirm (str) {
-          const results = Cypress.action('app:window:confirm', str)
+      // reset state back to empty object
+      this.state.reset()
 
-          // return false if ANY results are false
-          // else true
-          const ret = !_.some(results, returnedFalse)
+      // and then restore these backed up props
+      this.state(backup)
 
-          Cypress.action('app:window:confirmed', str, ret)
+      this.queue.reset()
+      this.queue.clear()
+      this.resetTimer()
+      this.testConfigOverride.restoreAndSetTestConfigOverrides(test, this.Cypress.config, this.Cypress.env)
 
-          return ret
-        },
-      })
+      this.removeAllListeners()
+    } catch (err) {
+      this.fail(err)
     }
+  }
 
-    const enqueue = function (obj) {
-      // if we have a nestedIndex it means we're processing
-      // nested commands and need to insert them into the
-      // index past the current index as opposed to
-      // pushing them to the end we also dont want to
-      // reset the run defer because splicing means we're
-      // already in a run loop and dont want to create another!
-      // we also reset the .next property to properly reference
-      // our new obj
-
-      // we had a bug that would bomb on custom commands when it was the
-      // first command. this was due to nestedIndex being undefined at that
-      // time. so we have to ensure to check that its any kind of number (even 0)
-      // in order to know to insert it into the existing array.
-      let nestedIndex = state('nestedIndex')
-
-      // if this is a number, then we know we're about to insert this
-      // into our commands and need to reset next + increment the index
-      if (_.isNumber(nestedIndex)) {
-        state('nestedIndex', (nestedIndex += 1))
-      }
+  addCommandSync (name, fn) {
+    const cy = this
 
-      // we look at whether or not nestedIndex is a number, because if it
-      // is then we need to insert inside of our commands, else just push
-      // it onto the end of the queue
-      const index = _.isNumber(nestedIndex) ? nestedIndex : queue.length
+    cy[name] = function () {
+      return fn.apply(cy.runnableCtx(name), arguments)
+    }
+  }
 
-      queue.insert(index, $Command.create(obj))
+  addChainer (name, fn) {
+    // add this function to our chainer class
+    return $Chainer.add(name, fn)
+  }
 
-      return Cypress.action('cy:command:enqueued', obj)
-    }
+  addCommand ({ name, fn, type, prevSubject }) {
+    const cy = this
 
-    const getCommandsUntilFirstParentOrValidSubject = function (command, memo = []) {
-      if (!command) {
-        return null
-      }
+    // TODO: prob don't need this anymore
+    this.commandFns[name] = fn
 
-      // push these onto the beginning of the commands array
-      memo.unshift(command)
+    const wrap = function (firstCall) {
+      fn = cy.commandFns[name]
```

Comment:
```suggestion
      fn = this.commandFns[name]
```

Reply:
It should be `cy` because `wrap` function is not an arrow function. When we use `this`, `this` tries to find the value connected with `wrap`, not the instance of `class $Cy`.